repository/src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Dialect/node-common-SqlMap.xml

```diff
@@ -953,7 +953,47 @@
             ca.child_node_id
         ]]>
     </sql>
-    
+
+    <select id="select_allChildAssocsOfParent" parameterType="ChildAssoc" resultMap="result_ChildAssoc">
+        <include refid="alfresco.node.select_ChildAssocsOfParentWithRecursive"></include>
+        <include refid="alfresco.node.select_ChildAssocsOfParent_Snippet"></include>
+        <include refid="alfresco.node.select_ChildAssocsOfParentRecursiveQuery"></include>
+    </select>
+    <sql id="select_ChildAssocsOfParentWithRecursive">
+        WITH RECURSIVE childAssosList AS (
+    </sql>
+    <sql id="select_ChildAssocsOfParentRecursiveQuery">
+        UNION ALL
+
+        select assoc.id                  as id,

```

Comment:
I'm concerned about what this returns. If we're adding a node to the root of the repository then does it return the whole repository?  What if we move a folder near the root of the repository to a new location?